ERCS/erc-7786.md

```diff
@@ -0,0 +1,184 @@
+
+MUST revert with `UnsupportedAttribute` if an unsupported attribute key is included. MAY revert if the value of an attribute is not a valid encoding for its expected type.
+
+MAY accept call value (native token) to be sent with the message. MUST revert if call value is included but it is not a feature supported by the gateway. It is unspecified how this value is represented on the destination.
+
+MAY generate and return a unique non-zero _outbox identifier_, otherwise returning zero. This identifier can be used to track the lifecycle of the message in the outbox in events and for post-processing.
+
+MUST emit a `MessagePosted` event, including the optional outbox identifier that is returned by the function.
+
+#### `MessagePosted`
+
+This event signals that a would-be sender has requested a message to be sent.
+
+If `outboxId` is present, post-processing MAY be required to send the message through the cross-chain channel.
+
+#### Post-processing
+
+After a sender has invoked `sendMessage`, further action MAY be required by the gateways to make the message effective. This is called _post-processing_. For example, some payment is typically required to cover the gas of executing the message at the destination.
+
+The exact interface for any such action is out of scope of this ERC. If the `postProcessingOwner` attribute is supported and present, such actions MUST be restricted to the specified account, otherwise they MUST be able to be performed by any party in a way that MUST NOT be able to compromise the eventual receipt of the message.
+
+### Reception Procedure
+
+A **Destination Gateway** is a contract that implements a protocol to validate messages sent on other chains. The interface of the destination gateway and how it is invoked is out of scope of this ERC.
+
+The protocol MUST ensure delivery of a sent message to its **receiver** using the `IERC7786Receiver` interface (specified below), which the receiver MUST implement.
+
+Once the message can be safely delivered (see Properties), the gateway MUST invoke `executeMessage` with the message contents.
+
+The gateway MUST verify that `executeMessage` returns the correct value, and MUST revert otherwise.
+
+```solidity
+interface IERC7786Receiver {
+    function executeMessage(
+        string calldata sourceChain, // CAIP-2 chain identifier
+        string calldata sender, // CAIP-10 account address
+        bytes calldata payload,
+        bytes[] calldata attributes
+    ) external payable returns (bytes4);
+}
+```
+
+#### `executeMessage`
+
+Delivery of a message sent from another chain.
+
+The receiver MUST validate that the caller of this function is a **known gateway**, i.e., one whose underlying cross-chain messaging protocol it trusts.
+
+MUST return `IERC7786Receiver.executeMessage.selector` (`0x675b049b`). 
+
+#### Interaction Diagram
+
+![](../assets/erc-7786/send-execute.svg)
```

Comment:
HTMLProfer fails because it cannot find that file.

@abcoathup can you help us with that part ?

Reply:
Looking at other ERCs it sounds like this might work? Don't know if you've tried it.

```suggestion
![](../assets/eip-7786/send-execute.svg)
```